packages/scoutgame-ui/src/components/projects/CreateProject/CreateProjectForm.tsx

```diff
@@ -0,0 +1,210 @@
+'use client';
+
+import { log } from '@charmverse/core/log';
+import { yupResolver } from '@hookform/resolvers/yup';
+import LoadingButton from '@mui/lab/LoadingButton';
+import { Button, Divider, FormLabel, Stack, TextField, Typography } from '@mui/material';
+import { createScoutProjectAction } from '@packages/scoutgame/projects/createScoutProjectAction';
+import type { CreateScoutProjectFormValues } from '@packages/scoutgame/projects/createScoutProjectSchema';
+import { createScoutProjectSchema } from '@packages/scoutgame/projects/createScoutProjectSchema';
+import { useRouter } from 'next/navigation';
+import { useAction } from 'next-safe-action/hooks';
+import { useEffect, useState } from 'react';
+import type { FieldErrors } from 'react-hook-form';
+import { Controller, useForm } from 'react-hook-form';
+import { useAccount } from 'wagmi';
+
+import { useMdScreen } from '../../../hooks/useMediaScreens';
+import { useUser } from '../../../providers/UserProvider';
+import { FormErrors } from '../../common/FormErrors';
+
+import { ProjectAvatarField } from './ProjectAvatarField';
+import { ProjectSmartContractForm } from './ProjectSmartContractForm';
+import { ProjectTeamMemberForm } from './ProjectTeamMemberForm';
+
+export function CreateProjectForm({ onCancel }: { onCancel: VoidFunction }) {
+  const { address } = useAccount();
+  const { user } = useUser();
+  const isMdScreen = useMdScreen();
+  const [errors, setErrors] = useState<string[] | null>(null);
+  const {
+    control,
+    getValues,
+    formState: { isDirty },
+    handleSubmit,
+    setValue
+  } = useForm({
+    resolver: yupResolver(createScoutProjectSchema),
+    mode: 'onChange',
+    defaultValues: {
+      avatar: '',
+      name: '',
+      description: '',
+      website: '',
+      github: '',
+      teamMembers: user
+        ? [{ scoutId: user.id, role: 'owner', avatar: user.avatar ?? '', displayName: user.displayName }]
+        : [],
+      contracts: [],
+      deployers: address ? [{ address, verifiedAt: new Date() }] : []
```

Comment:
we should probably assume address is always undefined to begin with

Reply:
yes thats the plan for v2